content/en/docs/reference/using-api/cel.md

```diff
@@ -0,0 +1,285 @@
+|-------------------|----------------------------------------------------------------------------------------------|
+| `__underscores__` | `__`                                                                                         |
+| `__dot__`         | `.`                                                                                          |
+| `__dash__`        | `-`                                                                                          |
+| `__slash__`       | `/`                                                                                          |
+| `__{keyword}__`   | [CEL **RESERVED** keyword](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#syntax) |
+
+When you escape any of CEL's **RESERVED** keywords you need to match the exact property name
+use the underscore escaping
+(for example, `int` in the word `sprint` would not be escaped and nor would it need to be).
+
+Examples on escaping:
+
+| property name | rule with escaped property name   |
+|---------------|-----------------------------------|
+| `namespace`   | `self.__namespace__ > 0`          |
+| `x-prop`      | `self.x__dash__prop > 0`          |
+| `redact__d`   | `self.redact__underscores__d > 0` |
+| `string`      | `self.startsWith('kube')`         |
+
+## Resource Constraints
+
+CEL is non-Turing complete and offers a variety of production safety controls to limit execution time. CEL's _resource constraint_ features provide feedback to developers about expression complexity and help protect the API server from excessive resource consumption during evaluation.
+CEL's _resource constraint_ features are used to prevent CEL evaluation from
+consuming excessive API server resources.
+
+### Runtime Cost Budget
+
+All CEL expressions evaluated by Kubernetes are constrained by a runtime cost
+budget. If the CEL interpreter executes too many instructions, the runtime cost
+budget will be exceeded, execution of the expressions will be halted, and an
+error will result.
+
+The runtime cost budget is an estimate of actual CPU utilization computed by
+incrementing a "cost unit" counter while interpreting a CEL expression. "Cost
+units" are independant of system load and hardware. Cost is also deterministic;
+for any given CEL expression and input data, evaluation of the expression by
+the CEL interpreter will always result in the same cost.
+
+Some Kubernetes resources may define a an additional runtime cost budget that
+bounds the execution of multiple expressions. If the sum total of the cost of
+expressions exceed the budget, execution of the expressions will be halted, and
+an error will result. For example the validation of a custom resource has a
+"per-valiation" runtime cost budget for all [Validation
+Rules](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules)
+evaluated to validate the custom resource.
```

Comment:
It would be nice to document whether or not webhook authz mode has an impact on runtime cost (I feel that it should - but if it doesn't then maybe we ought instead to warn users about the combination of CEL uses cases that allow authz checks, and webhook mode authz).

Reply:
Good catch. Authz mode does impact runtime cost. I'll clarify in this documentation. I also talked with sig-auth and was informed that the authz webhook calls, for the admission request principal, are already cached at this point, so CEL won't be impacted by webhook latency.

Reply:
Follow up:

The cost for authz checks is very high now at 10% of the expression budget per check. For cases where all the authz checks are via webhook mode, this seems _close_ to what we want.  I'd be okay with moving it 25% or 30% to make it even more restrictive. Thoughts?

I've also added a short warning about webhooks in the "Authorization library" section.